yksgi/src/lib.rs

```diff
@@ -149,6 +168,271 @@ impl SGInterp {
         }
     }
 
+    // Generate frames from stackmap information after a guard failure. The new frames are stored
+    // inside some allocated memory whose pointer this function returns. The frames are then later
+    // `memcpy`ed to the actual stack by ykcapi::__ykrt_reconstruct_frames.
+    pub fn reconstruct_frames(&self, btmframeaddr: *mut c_void) -> *const c_void {
+        // Load the stackmap and text sections from the binary. The former we need to parse in the
+        // stackmaps. The latter is required to disassemble the binary in order to distinguish
+        // branches from calls.
+        // FIXME: Don't use current_exe.
+        let pathb = env::current_exe().unwrap();
+        let file = fs::File::open(&pathb.as_path()).unwrap();
+        let exemmap = unsafe { memmap2::Mmap::map(&file).unwrap() };
+        let object = object::File::parse(&*exemmap).unwrap();
+        let sec = object.section_by_name(".llvm_stackmaps").unwrap();
+        let text = object.section_by_name(".text").unwrap();
+
+        // Create a capstone dissassembler.
+        // so we can figure out the size of the instruction we
+        // return too in order to adjust the return address in our reconstructed stackmap.
```

Comment:
too -> to

Reply:
Fixed in [c45e5e6](https://github.com/ykjit/yk/pull/578/commits/c45e5e644b2bc6b43c9d456a8192a5cc3ddbaff8)